src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java

```diff
@@ -104,22 +101,34 @@ public CountingReadFilter makeReadFilter() {
     
     @Override
     public void onTraversalStart() {
-        SAMFileHeader header = getHeaderForSAMWriter();
+        header = getHeaderForSAMWriter();
         rnaReadTransform = REFACTOR_NDN_CIGAR_READS ? new NDNCigarReadTransformer() : ReadTransformer.identity();
         try {
             referenceReader = new CachingIndexedFastaSequenceFile(referenceArguments.getReferenceFile());
             GenomeLocParser genomeLocParser = new GenomeLocParser(getBestAvailableSequenceDictionary());
             outputWriter = createSAMWriter(OUTPUT, false);
-            overhangManager = new OverhangFixingManager(header, outputWriter, genomeLocParser, referenceReader, MAX_RECORDS_IN_MEMORY, MAX_MISMATCHES_IN_OVERHANG, MAX_BASES_TO_CLIP, doNotFixOverhangs);
+            overhangManager = new OverhangFixingManager(header, outputWriter, genomeLocParser, referenceReader, MAX_RECORDS_IN_MEMORY, MAX_MISMATCHES_IN_OVERHANG, MAX_BASES_TO_CLIP, doNotFixOverhangs, processSecondaryAlignments);
 
         } catch (FileNotFoundException ex) {
             throw new UserException.CouldNotReadInputFile(referenceArguments.getReferenceFile(), ex);
         }
     }
 
     @Override
-    public void apply(GATKRead read, ReferenceContext referenceContext, FeatureContext featureContext) {
-        splitNCigarRead(rnaReadTransform.apply(read),overhangManager);
+    protected void firstPassApply(GATKRead read, ReferenceContext bytes, FeatureContext featureContext) {
+        splitNCigarRead(rnaReadTransform.apply(read),overhangManager, true, header, processSecondaryAlignments);
+    }
+
+    @Override
+    protected void secondPassApply(GATKRead read, ReferenceContext bytes, FeatureContext featureContext) {
+        splitNCigarRead(rnaReadTransform.apply(read),overhangManager, true, header, processSecondaryAlignments);
+    }
+
+    // Activates writing in the manager, which destinguishes each pass
+    @Override
+    protected void afterFirstPass() {
+        super.afterFirstPass();
```

Comment:
Right, this is an artifact of having had to reinitialize the reads source and filtering, seems I forgot to change it back.